### PR TITLE
Create prime_factorization_fast.py

### PR DESCRIPTION
**Pull request by sourabhkv**
optimising prime factorization for speed

### Describe your change:
I am adding fast prime factorization , printing prime factors of a given number faster than https://github.com/sourabhkv/Python/blob/master/maths/prime_factors.py approx ~2-2.5x faster

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: maths/prime_factorization_fast.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+from math import ceil, sqrt
+from __future__ import annotations
```
Comment (rohan472000) [inline maths/prime_factorization_fast.py:2]: use `ruff . --fix` to fix all ruff issues.

**Review by rohan472000 — CHANGES_REQUESTED** [review]

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
1) Do you have a source/reference for your algorithm (e.g., a Wikipedia article detailing the algorithm)?
2) Could you add your algorithm to the existing `prime_factors.py` file? This would make it easier for users to compare the different algorithms and it'd also help us maintainers keep track of all of them.
3) Could you add a benchmark function of some kind to `prime_factors.py` comparing your algorithm to the existing one? You should be able to back up your claim that your algorithm is twice as fast as the existing algorithm with data.

File: maths/prime_factorization_fast.py (hunk 1 above)
Comment (sourabhkv) [reply maths/prime_factorization_fast.py:1]: Resolved [Here](https://github.com/TheAlgorithms/Python/pull/8920/commits/2590a893f828a5b8e7462c1aae1d28f53263098b)

**Comment by rohan472000** [comment]
Before pushing u should rectify ruff issues, checking issue by 'ruff .' and then 'ruff . -- fix' if any issue found.

**Comment by sourabhkv** [comment]
> 1. I was doing prime factorization of a number then I thought of optimizing code, my goal was to minimize computing , so 2 is only even prime so I made its separate case , thus skipping other even numbers, I noted that either the same prime is repeated or a higher comes next . Somewhat related to [this](https://en.wikipedia.org/wiki/Integer_factorization#:~:text=key%20cryptography%20insecure.-,Prime%20decomposition,-%5Bedit%5D)
> 2. Added [here](https://github.com/TheAlgorithms/Python/pull/8920/commits/f94e08df91b3ffd48dc860a4cd93cb787332cdcf)
> 3. Results 
```
E:\algos>python -u "e:\algos\primeproduct3.py"
enter number: 476456345256745634
[2, 11, 9883, 2191349448809]
primeproduct took 0.111612 seconds to execute.
[2, 11, 9883, 2191349448809]
prime_factors took 0.323508 seconds to execute.

E:\algos>python -u "e:\algos\primeproduct3.py"
enter number: 56484515618741
[7, 7, 61, 18897462569]
primeproduct took 0.015743 seconds to execute.
[7, 7, 61, 18897462569]
prime_factors took 0.043377 seconds to execute.

E:\algos>python -u "e:\algos\primeproduct3.py"       
enter number: 456156165453421
[139, 3281699032039]
primeproduct took 0.139832 seconds to execute.
[139, 3281699032039]
prime_factors took 0.385396 seconds to execute.

E:\algos>python -u "e:\algos\primeproduct3.py"
enter number: 5646513612545313314684564564
[2, 2, 31, 19427, 68447, 902141, 37959817459]
primeproduct took 0.095845 seconds to execute.
[2, 2, 31, 19427, 68447, 902141, 37959817459]
prime_factors took 0.215599 seconds to execute.

E:\algos>python -u "e:\algos\primeproduct3.py"
enter number: 72575237587451742753725874147
[3, 128923, 8429947, 22259323667066729]
primeproduct took 10.020662 seconds to execute.
[3, 128923, 8429947, 22259323667066729]
prime_factors took 32.319923 seconds to execute.
```
primeproduct is clearly faster

**Comment by sourabhkv** [comment]
> Before pushing u should rectify ruff issues, checking issue by `ruff .` and then `ruff . -- fix` if any issue found.

I don't know how to use it

**Comment by rohan472000** [comment]
> > Before pushing u should rectify ruff issues, checking issue by `ruff .` and then `ruff . -- fix` if any issue found.
> 
> I don't know how to use it , Can you guide me?

See [Contributing.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md) of this repo.

**Comment by sourabhkv** [comment]
I cant make it, sorry

**Comment by rohan472000** [comment]
wait ....if u check the ruff errors then u will see that one more .py file should be reformatted.
image
run the same ruff command for that file too and push both.
Ruff issue will be resolved and then look into test cases again whats the difference between expected and actual output.

**Comment by tianyizheng02** [comment]
Closing due to failing tests. It looks like whatever changes this PR makes changed the behavior of the algorithm so that other files that imported this file started failing their own tests.

**Pull request closed without merging**